crates/uv-build-backend/src/lib.rs

```diff
@@ -336,6 +344,51 @@ pub fn metadata(source_tree: &Path, metadata_directory: &Path) -> Result<String,
     Ok(dist_info_dir)
 }
 
+/// PEP 517 requires that the metadata directory from the prepare metadata call is identical to the
+/// build wheel call. This method performs a prudence check that `METADATA` and `entry_points.txt`
+/// match.
+fn check_metadata_directory(
+    source_tree: &Path,
+    metadata_directory: Option<&Path>,
+    pyproject_toml: &PyProjectToml,
+) -> Result<(), Error> {
+    let Some(metadata_directory) = metadata_directory else {
+        return Ok(());
+    };
+
+    let dist_info_dir = format!(
+        "{}-{}.dist-info",
+        pyproject_toml.name().as_dist_info_name(),
+        pyproject_toml.version()
+    );
+
+    let metadata = pyproject_toml
+        .to_metadata(source_tree)?
+        .core_metadata_format();
+    if fs_err::read_to_string(metadata_directory.join(&dist_info_dir).join("METADATA"))? != metadata
```

Comment:
Nit: I'd put the LHS in a local variable since this is a mouthful to fit in an `if`. And then tweak the names, e.g., `pyproject_metadata` and `direct_metadata` or some such.